app/forms/support_interface/application_forms/nationalities_form.rb

```diff
@@ -0,0 +1,71 @@
+module SupportInterface
+  module ApplicationForms
+    class NationalitiesForm
+      include ActiveModel::Model
+
+      attr_accessor(
+        :first_nationality, :second_nationality, :british, :irish, :other,
+        :other_nationality1, :other_nationality2, :other_nationality3, :nationalities,
+        :audit_comment
+      )
+
+      validates(
+        :first_nationality, :second_nationality, :other_nationality1, :other_nationality2, :other_nationality3,
+        inclusion: { in: NATIONALITY_DEMONYMS, allow_blank: true }
+      )
+      validates :audit_comment, presence: true
+      validate :candidate_provided_nationality
+
+      UK_AND_IRISH_NATIONALITIES = ['British', 'Welsh', 'Scottish', 'Northern Irish', 'Irish', 'English'].freeze
+
+      def self.build_from_application(application_form)
+        new(
+          application_form.build_nationalities_hash,
```

Comment:
Where is `build_nationalities_hash` defined?

Reply:
https://github.com/DFE-Digital/apply-for-teacher-training/blob/b82508ab2ec8a255c050f20ec5e6f11f5fb5023c/app/models/application_form.rb#L240